### PR TITLE
Merge

### PR DESCRIPTION
**Pull request by grwilson**
Upstream merge with the following conflicts:
```
tests/runfiles/linux.run
module/zfs/dsl_scan.c
module/zfs/btree.c
include/sys/btree.h
```

### CODE REVIEW

**Comment by grwilson** [comment]
[zfs-precommit](http://platform.jenkins.delphix.com/job/devops-gate/job/master/job/zfs-precommit/4881/)

File: module/zfs/dsl_scan.c [hunk 1]
```diff
@@ -2925,21 +2925,13 @@ scan_io_queue_fetch_ext(dsl_scan_io_queue_t *queue)
 		if (zfs_scan_issue_strategy == 1) {
 			return (range_tree_first(rt));
 		} else if (zfs_scan_issue_strategy == 2) {
-			/*
-			 * We need to get the original entry in the by_addr
-			 * tree so we can modify it.
-			 */
```
Comment (pcd1193182) [inline module/zfs/dsl_scan.c]: We should take the local version of dsl_scan.c, actually. These comments were added in https://github.com/delphix/zfs/pull/81/files , and I'll port them upstream shortly.

File: module/zfs/btree.c [hunk 2]
```diff
@@ -53,11 +53,17 @@ kmem_cache_t *zfs_btree_leaf_cache;
  * (while the asymptotic complexity of the other steps is the same, the
  * importance of the constant factors cannot be denied).
  */
-#ifdef ZFS_DEBUG
-int zfs_btree_verify_intensity = 5;
-#else
 int zfs_btree_verify_intensity = 0;
-#endif
```
Comment (pcd1193182) [inline module/zfs/btree.c]: We should take the local version of this change internally, if we want to keep debugging the btree changes fully. We don't have to; we don't do this for any other datastructures in zfs. Something to discuss.

**Comment by grwilson** [comment]
[ab-pre-push](http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/appliance-build-orchestrator-pre-push/2309/)

**Pull request merged**